### PR TITLE
Adding intermediate response handling to BOXAPIMultipartToJSONOperati…

### PR DESCRIPTION
**Pull request by jdizoglio**
…on which supports urlsession tasks transferring data in parts. Supports variable length data responses for file requests. Our current example is file information returned via file upload request, certain field requests will extend the data size causing url session to return data in sections. We append the data and wait until task completion before serializing the data.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jdizoglio has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m [hunk 1]
```diff
@@ -127,24 +138,7 @@ - (void)sessionTask:(NSURLSessionTask *)sessionTask
...
-- (void)sessionTask:(NSURLSessionTask *)sessionTask processIntermediateData:(NSData *)data
-{
-    @synchronized (self) {
-        if (data != nil) {
-            [self processResponseData:data];
-        }
-    }
-}
-
-- (void)sessionTask:(NSURLSessionTask *)sessionTask didFinishWithResponse:(NSURLResponse *)response responseData:(nullable NSData *)responseData error:(NSError *)error
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m]: Are you sure you can take this out?? That might be a problem

Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m]: merge issue.

**Review by boxtcarpel — APPROVED** [review]

**Comment by thuyBox** [comment]
+2

**Pull request merged**